corepkgs/buildenv.pl

```diff
@@ -110,12 +110,13 @@ sub addPkg {
     my $propagatedFN = "$pkgDir/nix-support/propagated-user-env-packages";
     if (-e $propagatedFN) {
         open PROP, "<$propagatedFN" or die;
-        my $propagated = <PROP>;
-        close PROP;
-        my @propagated = split ' ', $propagated;
-        foreach my $p (@propagated) {
-            $postponed{$p} = 1 unless defined $done{$p};
+        while (my $propagated = <DATA>) {
```

Comment:
@edolstra sounds good to me. That exactly mirrors what `stdenv/setup.sh` did before my changes.